test_files/type/type.js

```diff
@@ -18,8 +18,8 @@ let /** @type {!Array<?>} */ typeTupleTuple = [[1, 2]];
 let /** @type {!Array<?>} */ typeTupleTuple2 = [[1, 2], ''];
 let /** @type {(string|boolean)} */ typeUnion = Math.random() > 0.5 ? false : '';
 let /** @type {(string|boolean)} */ typeUnion2 = Math.random() > 0.5 ? false : '';
-let /** @type {{optional: (undefined|boolean)}} */ typeOptionalField = {};
-let /** @type {{optional: (undefined|string|boolean)}} */ typeOptionalUnion = {};
+let /** @type {{optional: ((undefined|boolean)|undefined)}} */ typeOptionalField = {};
```

Comment:
`undefined|boolean|undefined`?

Reply:
Yeah. Doesn't hurt, and makes the code easier/more consistent.